frontend/online-shop/src/services/products.service.tsx

```diff
@@ -0,0 +1,26 @@
+import axios from "axios";
+
+import { productsEndpoint } from "../helpers/strings";
+
+export const fetchProducts = ({ setProduct }: any) => {
+  axios.get(productsEndpoint).then(
```

Comment:
Instead of using callbacks you should use `await` and return the response within this function. This way you can also do what I mentioned earlier, to change the state within the component not in here. The purpose of this service is to only get the data from the backend and pass it on, that's it